general/git.md

```diff
@@ -0,0 +1,34 @@
+# Git & GitHub
+For source control we use GitHub. All projects are repositories under [scrambledeggs](https://github.com/scrambledeggs).
+
+## Workflow
+- Prefer to adapt [Gitlab flow](https://about.gitlab.com/2014/09/29/gitlab-flow/)
+- The canonical version of the code for active development lives in `master`
+- When a new feature is worked on, a branch will be created from `master`
+- Use descriptive feature branch names and name them using [`kebab-case`](https://en.wikipedia.org/wiki/Letter_case#Special_case_styles)
+- For fixes, treat the same as features but add `fix` to branch name i.e `fix-your-face`
+- `master` will be write protected and merging to `master` will require a [pull request](https://help.github.com/articles/about-pull-requests/).
+  ### Deployments and Releases
+  - For app submissions, squash commit to `release` branch and tag it with the version number
+  - For deployed apps, squash commit to `production` branch and tag it with version number when applicable
+
+## Versioning
+- Prefer to use [semantic versioning](http://semver.org/) where applicable.
+
+## Commits and Pull Requests
+- Commit often. Small commits are acceptable. General guideline is to commit when tests are passing.
+- Make your commits contextual. Avoid commits that does other things that the commit was intended to.
+- Commit messages that makes sense. Usually in this pattern: `"{action} {purpose|reason} {target}"`, order may vary.
+  Avoid:
```

Comment:
`master` is the "`dev`" branch. does that make sense? can we find a good example for this that has a good workflow